packages/in_app_purchase/in_app_purchase_storekit/darwin/Classes/InAppPurchasePlugin.swift

```diff
@@ -0,0 +1,428 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import Foundation
+import StoreKit
+
+#if os(iOS)
+  import Flutter
+#elseif os(macOS)
+  import FlutterMacOS
+#endif
+
+public class InAppPurchasePlugin: NSObject, FlutterPlugin, InAppPurchaseAPI {
+  private let receiptManager: FIAPReceiptManager
+  private var productsCache: NSMutableDictionary = [:]
+  private var paymentQueueDelegateCallbackChannel: FlutterMethodChannel?
+  // note - the type should be FIAPPaymentQueueDelegate, but this is only available >= iOS 13,
+  // FIAPPaymentQueueDelegateonly gets set/used in registerPaymentQueueDelegateWithError or removePaymentQueueDelegateWithError, which both are ios13+ only
+  private var paymentQueueDelegate: Any?
+  private var requestHandlers = Set<FIAPRequestHandler>()
+  private var handlerFactory: ((SKRequest) -> FIAPRequestHandler)
+  // TODO(louisehsu): Once tests are migrated to swift, we can use @testable import, and make theses vars private again and remove all instances of @objc
+  @objc
+  public var registrar: FlutterPluginRegistrar?
+  @objc
+  // This property is optional, as it requires self to exist to be initialized.
+  public var paymentQueueHandler: FIAPaymentQueueHandler?
+  @objc
+  // This property is optional, as it is set when the plugin is registered.
+  public var transactionObserverCallbackChannel: FlutterMethodChannel?
+
+  public static func register(with registrar: FlutterPluginRegistrar) {
+    #if os(iOS)
+      let messenger = registrar.messenger()
+    #endif
+    #if os(macOS)
+      let messenger = registrar.messenger
+    #endif
+    let channel = FlutterMethodChannel(
+      name: "plugins.flutter.io/in_app_purchase",
+      binaryMessenger: messenger)
+    let instance = InAppPurchasePlugin(registrar: registrar)
+    registrar.addMethodCallDelegate(instance, channel: channel)
+    registrar.addApplicationDelegate(instance)
+    SetUpInAppPurchaseAPI(messenger, instance)
+  }
+
+  @objc
+  public init(
+    receiptManager: FIAPReceiptManager,
+    handlerFactory: @escaping (SKRequest) -> FIAPRequestHandler = {
+      FIAPRequestHandler(request: $0)
+    }
+  ) {
+    self.receiptManager = receiptManager
+    self.requestHandlers = Set<FIAPRequestHandler>()
+    self.handlerFactory = handlerFactory
+    super.init()
+  }
+
+  // This init is used for tests
+  @objc
+  public convenience init(receiptManager: FIAPReceiptManager) {
+    self.init(receiptManager: receiptManager, handlerFactory: { FIAPRequestHandler(request: $0) })
```

Comment:
>   // This init is used for tests

Can we simply pass `FIAPRequestHandler(request: $0) }` in test?

Reply:
Oh, thats true. Separately, do you think there's merit in keeping this init, if this is the designated init from the objc version `initWithRecieptManager:`? It might be a breaking change if this is removed.

Reply:
tho im not sure how common it is to be initializing plugins manually?

Reply:
if it's only used in tests (as stated in the comment), then it's not gonna be a breaking change. I could be wrong, but I don't think this is used by developers. Can you double check?

Reply:
i did a quick search and couldnt find anything - so i just deleted it lol